presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/RowExpressionVerifier.java

```diff
@@ -0,0 +1,512 @@
+            return getValueFromLiteral(actual).equals(String.valueOf(rowExpressionInterpreter(expected, metadata, session).evaluate()));
+        }
+        if (expected instanceof ConstantExpression) {
+            return getValueFromLiteral(actual).equals(String.valueOf(LiteralInterpreter.evaluate(TEST_SESSION.toConnectorSession(), (ConstantExpression) expected)));
+        }
+        return false;
+    }
+
+    @Override
+    protected Boolean visitStringLiteral(StringLiteral actual, RowExpression expected)
+    {
+        if (expected instanceof ConstantExpression && expected.getType().getJavaType() == Slice.class) {
+            String expectedString = (String) LiteralInterpreter.evaluate(TEST_SESSION.toConnectorSession(), (ConstantExpression) expected);
+            return actual.getValue().equals(expectedString);
+        }
+        return false;
+    }
+
+    @Override
+    protected Boolean visitLogicalBinaryExpression(LogicalBinaryExpression actual, RowExpression expected)
+    {
+        if (expected instanceof SpecialFormExpression) {
+            SpecialFormExpression expectedLogicalBinary = (SpecialFormExpression) expected;
+            if ((actual.getOperator() == OR && expectedLogicalBinary.getForm() == SpecialFormExpression.Form.OR) ||
+                    (actual.getOperator() == AND && expectedLogicalBinary.getForm() == SpecialFormExpression.Form.AND)) {
+                return process(actual.getLeft(), expectedLogicalBinary.getArguments().get(0)) &&
+                        process(actual.getRight(), expectedLogicalBinary.getArguments().get(1));
+            }
+        }
+        return false;
+    }
+
+    @Override
+    protected Boolean visitBetweenPredicate(BetweenPredicate actual, RowExpression expected)
+    {
+        if (expected instanceof CallExpression && functionResolution.isBetweenFunction(((CallExpression) expected).getFunctionHandle())) {
+            return process(actual.getValue(), ((CallExpression) expected).getArguments().get(0)) &&
+                    process(actual.getMin(), ((CallExpression) expected).getArguments().get(1)) &&
+                    process(actual.getMax(), ((CallExpression) expected).getArguments().get(2));
+        }
+
+        return false;
+    }
+
+    @Override
+    protected Boolean visitNotExpression(NotExpression actual, RowExpression expected)
+    {
+        if (!(expected instanceof CallExpression) || !functionResolution.notFunction().equals(((CallExpression) expected).getFunctionHandle())) {
+            return false;
+        }
+        return process(actual.getValue(), ((CallExpression) expected).getArguments().get(0));
+    }
+
+    @Override
+    protected Boolean visitSymbolReference(SymbolReference actual, RowExpression expected)
+    {
+        if (!(expected instanceof VariableReferenceExpression)) {
+            return false;
+        }
+        return symbolAliases.get((actual).getName()).getName().equals(((VariableReferenceExpression) expected).getName());
```

Comment:
This means the SymbolAliases is a mapping from the symbol to the alias, rather than the other way around as original right? This is confusing.

Reply:
I was kinda lazy to flip symbolAliases; so I just directly flipped the expression rowExpression. But sure, we can flip the aliases to make it consistent.